src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -85,25 +87,40 @@ private Optional<EppoValue> getAssignmentValue(
             return Optional.empty();
         }
 
-        // Find matched rule
-        Optional<Rule> rule = RuleValidator.findMatchingRule(subjectAttributes, configuration.getRules());
-        if (rule.isEmpty()) {
-            log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
-            return Optional.empty();
-        }
-
-        // Check if in experiment sample
-        String allocationKey = rule.get().getAllocationKey();
-        Allocation allocation = configuration.getAllocation(allocationKey);
-        if (!this.isInExperimentSample(subjectKey, flagKey, configuration.getSubjectShards(),
-                allocation.getPercentExposure())) {
-            log.info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
-            return Optional.empty();
+        String allocationKey;
+        List<Variation> variations;
+        if (configuration.isBandit()) {
+            allocationKey = "bandit";
+            variations = BanditEvaluator.evaluateBanditVariations(
+              flagKey,
+              "random",
+              assignmentOptions,
+              subjectKey,
+              subjectAttributes,
+              configuration.getSubjectShards()
+            );
+        } else {
+            // Find matched rule
+            Optional<Rule> rule = RuleValidator.findMatchingRule(subjectAttributes, configuration.getRules());
+            if (rule.isEmpty()) {
+                log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
+                return Optional.empty();
+            }
+
+            // Check if in experiment sample
+            allocationKey = rule.get().getAllocationKey();
+            Allocation allocation = configuration.getAllocation(allocationKey);
+            if (!this.isInExperimentSample(subjectKey, flagKey, configuration.getSubjectShards(),
+              allocation.getPercentExposure())) {
+                log.info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
+                return Optional.empty();
+            }
+
+            variations = allocation.getVariations();
```

Comment:
This code was basically left unchanged, just indented and some declarations pulled out of the if-else block. GitHub diff, just a bit confused.